### PR TITLE
Resolve filenames in `backtrace` command

### PR DESCRIPTION
**Pull request by rocky**
Save relative to absolute file name mapping when doing "load" command, and in the debugger callback hook .

Resolve file paths in "backtrace" command.

### CODE REVIEW

**Comment by rocky** [comment]
I should add a test case for this.

File: command/backtrace.sh [hunk 1]
```diff
@@ -129,7 +129,8 @@ function _Dbg_do_backtrace {
     if (( frame_start == 0 )) ; then
 	((count--)) ;
 	adjusted_pos=$(_Dbg_frame_adjusted_pos 0)
-	filename=$(_Dbg_file_canonic "${BASH_SOURCE[$adjusted_pos]}")
+	filename=$(_Dbg_resolve_expand_filename "${BASH_SOURCE[$adjusted_pos]}")
```
Comment (jansorg) [inline command/backtrace.sh]:
```suggestion
	filename="$(_Dbg_resolve_expand_filename "${BASH_SOURCE[$adjusted_pos]}")"
```

File: command/backtrace.sh [hunk 2]
```diff
@@ -129,7 +129,8 @@ function _Dbg_do_backtrace {
...
+	filename=$(_Dbg_adjust_filename "$filename")
```
Comment (jansorg) [inline command/backtrace.sh]:
```suggestion
	filename="$(_Dbg_adjust_filename "$filename")"
```

File: command/backtrace.sh [hunk 3]
```diff
@@ -170,7 +169,8 @@ function _Dbg_do_backtrace {
 	else
 	    lineno=${BASH_LINENO[$adjusted_pos-1]}
 	fi
-	filename=$(_Dbg_file_canonic "${BASH_SOURCE[$adjusted_pos]}")
+	filename=$(_Dbg_resolve_expand_filename "${BASH_SOURCE[$adjusted_pos]}")
```
Comment (jansorg) [inline command/backtrace.sh]:
```suggestion
	filename="$(_Dbg_resolve_expand_filename "${BASH_SOURCE[$adjusted_pos]}")"
```

File: command/backtrace.sh [hunk 4]
```diff
@@ -170,7 +169,8 @@ function _Dbg_do_backtrace {
...
+	filename=$(_Dbg_adjust_filename "$filename")
```
Comment (jansorg) [inline command/backtrace.sh]:
```suggestion
	filename="$(_Dbg_adjust_filename "$filename")"
```

File: lib/file.sh [hunk 5]
```diff
@@ -105,14 +105,14 @@ function _Dbg_resolve_expand_filename {
...
     typeset -i i
     for (( i=0 ; i < n; i++ )) ; do
-      typeset basename="${_Dbg_dir[i]}"
-      if [[  "$basename" == '\$cdir' ]] ; then
-	basename=$_Dbg_cdir
-      elif [[ "$basename" == '\$cwd' ]] ; then
-	basename=$(pwd)
+      typeset dirname="${_Dbg_dir[i]}"
+      if [[  "$dirname" == '\$cdir' ]] ; then
+	dirname=$_Dbg_cdir
```
Comment (jansorg) [inline lib/file.sh]:
```suggestion
	dirname="$_Dbg_cdir"
```

File: lib/file.sh [hunk 6]
```diff
@@ -105,14 +105,14 @@ function _Dbg_resolve_expand_filename {
...
+      elif [[ "$dirname" == '\$cwd' ]] ; then
+	dirname=$(pwd)
```
Comment (jansorg) [inline lib/file.sh]:
```suggestion
	dirname="$(pwd)"
```

**Review by jansorg — APPROVED** [review]
@rocky Thank you very much!
This PR works well for my first use-case.
I may be too picky with quotes, but I've added a few code suggestions for quoting. Some were already in the code, I think.

**Comment by rocky** [comment]
> @rocky Thank you very much! This PR works well for my first use-case.

Good to hear! Thanks for the confirmation.

>  I may be too picky with quotes, but I've added a few code suggestions for quoting. Some were already in the code, I think.

I appreciate the corrections. Let's get this as good as we can.

**Pull request merged**